### ISSUE TITLE
Bonify Group Experience

### ISSUE THREAD

**Issue by helenatodev** [AscensionGameDev/Intersect-Engine#749]
**Is your feature request related to a problem? Please describe.**
What do you think of a bonus in XP gain for each member of the group?

**Describe the solution you'd like**
Currently a monster's XP is divided ALSO for each member of the group. But for some members that would be a little unfair.
What is the benefit of going with 4 friends to kill 1 monster, where your EXP will be divided equally for everyone?

I believe that this would better influence the groups, make them have an advantage.

But of course, this would be optional in the server settings, and also the percentage of the gain to be increased ...

**Additional context**
And I come here to ask everyone, what do you think of that?
If they approve, I can send a PR with the changes.

**Comment by WeylonSantana** [comment]
How would it work? The xp would no longer be divided, but increased by a percentage? Would we have the option to disable the xp division and exchange it for a bonus increase for each member? Would the xp bonus be made on top of the individual xp already divided by each member of the group? Could there be different values for each member or would it be a global value?

**Comment by dhmhc** [comment]
I will answer in an example that will be easier for everyone to understand how the system will work.
Percentage configured on the gain server.
Ex: 10 = 10%
Monster gives 1000 XP.
1 player, alone, without a party, will win 1000 by killing him.
2 players in a party, will kill this monster.
The Exp that will be divided will be +20%
Then, the monster will now divide 1200 XP, totaling a division of 600 XP for each member.
In short, for each additional member in a group you will gain + 10% xp on top of the xp to be divided for EVERYONE.
The particular EXP I cannot change because there is already a configuration in the items that can add XP individual bonus for each one.

**Comment by da-rulez** [comment]
Seems a little bit OP to me. You already have the advantage that you can kill the NPC a lot faster, so extra XP on top of that would be a double bonus. 

I personally wouldn't use this functionality but it can't hurt if you make it an option in the NPC editor I suppose?

**Comment by dhmhc** [comment]
> Seems a little bit OP to me. You already have the advantage that you can kill the NPC a lot faster, so extra XP on top of that would be a double bonus.
> 
> I personally wouldn't use this functionality but it can't hurt if you make it an option in the NPC editor I suppose?

Making it available in the NPC editor would make it more complicated when configuring, and this is currently a global configuration.
You can use it for example in a BOSS, this option of you gaining more XP if you are in a group can gather players to do it.

As said it is a function that may or may not be part of your project, as well as many other things that already exist in the engine, it is up to you whether you want to use it or not, as the engine is something made for several projects, this can be very useful for someone 😄 

We have to analyze it as something for EVERYONE ❤️  and not just our private projects, I already have the code ready, tested and working perfectly and very simple to understand.

**Comment by da-rulez** [comment]
Yes exactly. Everyone. That's why I'm saying I'm one of the people who wouldn't use it, and with me others. 

Like I said. Fine if you make it optional, but it doesn't seem like something that would fit in the base engine.

**Comment by AisenGaming** [comment]
This is for sure something that could be both a good and a bad thing.. It really comes down to the individual game design. It helps incentivize people to form parties in order to earn more exp but at the same time this could be a detrimental effect for those who like to solo challenge their way through an MMO.
So I guess I both approve and disapprove of this.

**Comment by Spectreye** [comment]
I think it's a good idea, but it's better as an option in the npc editor rather than a global option in server config. 

It would allow to configure it with more precision.

**Comment by RandemGamor** [comment]
**Party EXP Bonuses work well if they work like this for example;**

EXP / number of party members.
+ 5% additional exp for every party member.

So let's say.. Monster gives 100 Exp.
5 party members.
That's 20 Exp per member.
+ 5% per member is +25% total.
So that would be 25 Exp per member.
Total exp is then 125, shared across 5 people.

Which is a nice bonus if people are pulling their weight.

If this was added then that bonus % per member should be configurable via a json file.

To get full EXP from a monster for every player (would be OP) then all a person would need to do is set the % to 100 per player.
So then if 100 Exp for 1 monster.
/ 5 players.
= 20 Exp.
100% x 5 players = 500%
20 Exp x 500% = 100 Exp (As normal)

I would really love for this to be added.

**Comment by dhmhc** [comment]
> **Party EXP Bonuses work well if they work like this for example;**
> 
> EXP / number of party members.
> +5% additional exp for every party member.
> 
> So let's say.. Monster gives 100 Exp.
> 5 party members.
> That's 20 Exp per member.
> +5% per member is +25% total.
> So that would be 25 Exp per member.
> Total exp is then 125, shared across 5 people.
> 
> Which is a nice bonus if people are pulling their weight.
> 
> If this was added then that bonus % per member should be configurable via a json file.
> 
> To get full EXP from a monster for every player (would be OP) then all a person would need to do is set the % to 100 per player.
> So then if 100 Exp for 1 monster.
> / 5 players.
> = 20 Exp.
> 100% x 5 players = 500%
> 20 Exp x 500% = 100 Exp (As normal)
> 
> I would really love for this to be added.
> 
> [Edit: Read above comments properly now, basically what the OP wants
> I've just broken it down a little better I think :D]

Yes, that's exactly what I explained, the bonus percentage value can already be configured on the server.
The extra percentage would be based on the mob's total xp, divided to all members equally.

**Comment by Arufonsu** [comment]
This is not a bad idea, these days many mmos are ghost towns due to lack of incentives like these (its more worthy to solo everything, thus the community avoid social interactions and party play). 

This is helpful for devs that are looking foward to make their communities to play the game together, and get rewarded because of it 👍🏽 +1, but of course, since the idea of an open source engine like this is to be as dynamic and neutral as possible, this should be an optional feature for the game designer to choose.

**Comment by dhmhc** [comment]
> This is not a bad idea, these days many mmos are ghost towns due to lack of incentives like these (its more worthy to solo everything, thus the community avoid social interactions and party play).
> 
> This is helpful for devs that are looking foward to make their communities to play the game together, and get rewarded because of it 👍🏽 +1, but of course, since the idea of an open source engine like this is to be as dynamic and neutral as possible, this should be an optional feature for the game designer, thus not a default feature, this way the dev may choose whether or not toggle this feature on.

Exactly what I want people to understand, like it or not, it would be possible to configure. I already have the Pr ready for this, as soon as the new version comes out with my other pr, I'll send the bonus
